rom/dev/src/flow/cold_reset/fw_processor.rs

```diff
@@ -106,52 +121,81 @@ impl FirmwareProcessor {
         })
     }
 
-    /// Download the image
+    /// Process mailbox commands
     ///
     /// # Arguments
     ///
-    /// * `env` - ROM Environment
+    /// * `soc_ifc` - SOC Interface
+    /// * `mbox` - Mailbox
+    /// * `pcr_bank` - PCR Bank
+    /// * `sha384` - SHA384
     ///
     /// # Returns
+    /// * `MailboxRecvTxn` - Mailbox Receive Transaction
     ///
     /// Mailbox transaction handle. This transaction is ManuallyDrop because we
     /// don't want the transaction to be completed with failure until after
     /// handle_fatal_error is called. This prevents a race condition where the SoC
     /// reads FW_ERROR_NON_FATAL immediately after the mailbox transaction
     /// fails, but before caliptra has set the FW_ERROR_NON_FATAL register.
-    fn download_image<'a>(
+    fn process_mailbox_commands<'a>(
         soc_ifc: &mut SocIfc,
         mbox: &'a mut Mailbox,
+        pcr_bank: &mut PcrBank,
+        sha384: &mut Sha384,
     ) -> CaliptraResult<ManuallyDrop<MailboxRecvTxn<'a>>> {
         soc_ifc.flow_status_set_ready_for_firmware();
 
-        cprint!("[afmc] Waiting for Image ");
+        cprint!("[afmc] Waiting for Measurement(s) or Image ");
+        let mut measurement_count = 0;
         loop {
             if let Some(txn) = mbox.peek_recv() {
-                if txn.cmd() != Self::MBOX_DOWNLOAD_FIRMWARE_CMD_ID {
-                    cprintln!("Invalid command 0x{:08x} received", txn.cmd());
-                    txn.start_txn().complete(false)?;
-                    continue;
+                match txn.cmd() {
+                    Self::MBOX_STASH_MEASUREMENT_CMD_ID => {
+                        if measurement_count == Self::MEASUREMENT_MAX_COUNT {
+                            cprintln!(
+                                "[afmc] Maximum supported number of measurements already received, ignoring."
+                            );
+                            txn.start_txn().complete(false)?;
+                            continue;
+                        }
+
+                        // Re-borrow mailbox to work around https://github.com/rust-lang/rust/issues/54663
+                        let txn = mbox
+                            .peek_recv()
+                            .ok_or(CaliptraError::FW_PROC_MAILBOX_STATE_INCONSISTENT)?;
+
+                        let mut txn = ManuallyDrop::new(txn.start_txn());
+                        Self::extend_measurement(pcr_bank, sha384, &mut txn)?;
+                        measurement_count += 1;
+                        txn.complete(true)?;
```

Comment:
We should be writing the response as well right? (See readme at https://github.com/chipsalliance/caliptra-sw/tree/main/runtime). dpe_status and fips_status can always return success I think.

Reply:
Just a note here I am planning on adding the FIPS status to the response once this change is in. I'm not familiar with the DPE status but if it's simple I can probably handle that in the same PR.

Also the mailbox driver used in ROM currently does not support sending a response. I am working with Anthony on adding that separately right now.

Reply:
https://github.com/chipsalliance/caliptra-sw/pull/734 will address the ROM response .